app/uk/gov/hmrc/eoricommoncomponent/frontend/services/RegistrationDetailsService.scala

```diff
@@ -42,13 +44,9 @@ class RegistrationDetailsService @Inject() (sessionCache: SessionCache)(implicit
   def initialiseCacheWithRegistrationDetails(
     organisationType: CdsOrganisationType
   )(implicit request: Request[_]): Future[Boolean] =
-    sessionCache.subscriptionDetails flatMap { subDetails =>
-      sessionCache.saveSubscriptionDetails(
-        subDetails.copy(formData = subDetails.formData.copy(organisationType = Some(organisationType)))
-      )
-
-      saveRegistrationDetails(organisationType)
-    }
+    sessionCache.saveSubscriptionDetails(
+      SubscriptionDetails(formData = FormData(organisationType = Some(organisationType)))
+    ).flatMap(_ => saveRegistrationDetails(organisationType))
```

Comment:
Considering the ticket, lets have tests to prove that the cache is in fact cleared.